src/cell.cpp

```diff
@@ -235,5 +236,49 @@ bool cell_basic_equality(cell const& lhs, cell const& rhs)
     return true;
 }
 
+// Construct cell from flat morphology specification.
+
+cell make_cell(const morphology& morph) {
+    using point3d = cell::point_type;
+    cell newcell;
+
+    if (!morph) {
+        return newcell;
+    }
+
+    EXPECTS(morph.check_invariants());
+
+    // (not supporting soma-less cells yet)
+    newcell.add_soma(morph.soma.r, point3d(morph.soma.x, morph.soma.y, morph.soma.z));
+
+    for (const section_geometry& section: morph.sections) {
+        segmentKind kind = segmentKind::none;
```

Comment:
The problem here is that `section_kind` (from the morphology) and `segmentKind` (from the cell representation) are different enumerations; we can't assign directly safely.

Using morphology `section_kind` for `segmentKind` in cells would be a cleaner way of doing things, side-stepping the issue; I only refrained from doing that to make the changes (slightly) less invasive.